imap_processing/ccsds/excel_to_xtce.py

```diff
@@ -0,0 +1,426 @@
+        encoding : Element
+            The encoding element to add the conversion to.
+        """
+        # Look up the conversion in the AnalogConversions tab
+        analog_conversion = self.sheets["AnalogConversions"]
+        # conversion is a row from the AnalogConversions sheet
+        conversion = analog_conversion.loc[
+            (analog_conversion["mnemonic"] == row["mnemonic"])
+            & (analog_conversion["packetName"] == row["packetName"])
+        ].iloc[0]
+
+        # Create the Conversion element
+        default_calibrator = Et.SubElement(encoding, "xtce:DefaultCalibrator")
+        polynomial_calibrator = Et.SubElement(
+            default_calibrator, "xtce:PolynomialCalibrator"
+        )
+        # FIXME: Use lowValue / highValue from the conversion sheet
+        # FIXME: Handle segmented polynomials (only using first segment now)
+        for i in range(8):
+            col = f"c{i}"
+            if conversion[col] != 0:
+                term = Et.SubElement(polynomial_calibrator, "xtce:Term")
+                term.attrib["coefficient"] = str(conversion[col])
+                term.attrib["exponent"] = str(i)
+
+    def to_xml(self, output_xml_path: Path) -> None:
+        """
+        Create and output an XTCE file from the Element Tree representation.
+
+        Parameters
+        ----------
+        output_xml_path : Path
+            Path to the output XML file.
+        """
+        # Create the XML tree and save the document
+        tree = Et.ElementTree(self._root)
+        Et.indent(tree, space="\t", level=0)
+
+        # Use the provided output_xml_path
+        tree.write(output_xml_path, encoding="utf-8", xml_declaration=True)
+
+
+# Function to parse command line arguments
+def _parse_args() -> argparse.Namespace:
+    """
+    Parse the command line arguments.
+
+    The expected input format is a required argument of "/path/to/excel_file.xlsx"
+    with an optional argument containing the output path for the XTCE file
+    "/path/to/output.xml".
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values.
+    """
+    description = (
+        "This command line program generates an instrument specific XTCE file. "
+        "Example usage: imap_xtce "
+        "path/to/excel_packet_file.xlsx --output path/to/output_packet_definition.xml"
+    )
+    output_help = (
+        "Where to save the output XTCE file. "
+        "If not provided, the input file name will be used with a "
+        ".xml extension."
+    )
+    file_path_help = "Provide the full path to the input excel file."
+
+    parser = argparse.ArgumentParser(prog="imap_xtce", description=description)
+    parser.add_argument("excel_file", type=Path, help=file_path_help)
+    parser.add_argument("--output", type=Path, required=False, help=output_help)
+
+    if not find_spec("openpyxl"):
+        parser.error(
```

Comment:
Does it make sense to just include openpyxl in the imap_processing installer? I understand wanting to keep it small, but I don't love this. However, I can live with it if openpyxl is a huge dependency or something.

Reply:
This is somewhat a personal preference of mine where if it isn't going to be used a majority of the time, then we probably don't need it as a direct dependency. It is relatively small, but this should really only be used once to generate the XTCE file I think. Once we have the XTCE file, then the dependency isn't ever used again.

I didn't really want to bring this into the main package and just leave it in the "tools" section, but then the CLI that @anamanica added wouldn't be installed with the package because the tools/ directory isn't installed.

Not a great answer for you, but I'd prefer to keep it optional and not a required dependency.